### PR TITLE
posy-cursors: init at 1.6

### PR DESCRIPTION
**Pull request by mk3z**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (nagy) [inline maintainers/maintainer-list.nix:1]: The changes in this file should be in a separate commit. You can find good examples of the commit message in the history of that file https://github.com/NixOS/nixpkgs/commits/master/maintainers/maintainer-list.nix .

File: pkgs/data/icons/posy-cursors/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib, stdenvNoCC, fetchgit }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "posy-cursors";
+  version = "1.6";
+
+  src = fetchgit {
```
Comment (nagy) [inline pkgs/data/icons/posy-cursors/default.nix:7]: can you use `fetchFromGitHub` ?

File: pkgs/data/icons/posy-cursors/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  installPhase = ''
+    install -dm 755 $out/share/icons
+    cp -r Posy_Cursor* $out/share/icons
+  '';
+
+  meta = with lib; {
+    description = "Posy's Improved Cursors for Linux";
+    homepage = "https://github.com/simtrami/posy-improved-cursor-linux";
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ mkez ];
```
Comment (nagy) [inline pkgs/data/icons/posy-cursors/default.nix:22]: Also, we need to know the license of the project. I cant find any on the repo site.

File: pkgs/data/icons/posy-cursors/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "posy-cursors";
+  version = "1.6";
+
+  src = fetchgit {
+    url = "https://github.com/simtrami/posy-improved-cursor-linux";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-i0N/QB5uzqHapMCDl6h6PWPJ4GOAyB1ds9qlqmZacLY=";
+  };
+
+  installPhase = ''
```
Comment (nagy) [inline pkgs/data/icons/posy-cursors/default.nix:13]: Add `runHook preInstall` and `runHook postInstall` respectively.

File: pkgs/data/icons/posy-cursors/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    install -dm 755 $out/share/icons
```
Comment (nagy) [inline pkgs/data/icons/posy-cursors/default.nix:14]: You can use `mkdir -p` here.

File: pkgs/data/icons/posy-cursors/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ lib, stdenvNoCC, fetchgit }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "posy-cursors";
+  version = "1.6";
+
+  src = fetchgit {
+    url = "https://github.com/simtrami/posy-improved-cursor-linux";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-i0N/QB5uzqHapMCDl6h6PWPJ4GOAyB1ds9qlqmZacLY=";
```
Comment (nagy) [inline pkgs/data/icons/posy-cursors/default.nix:10]: Can you use `hash` ?

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/data/icons/posy-cursors/default.nix (hunk 5 above)
Comment (mk3z) [reply pkgs/data/icons/posy-cursors/default.nix:10]: Do you mean a commit hash?

File: pkgs/data/icons/posy-cursors/default.nix (hunk 2 above)
Comment (mk3z) [reply pkgs/data/icons/posy-cursors/default.nix:22]: Sadly the original creator has not specified a license: [http://www.michieldb.nl/other/cursors/](http://www.michieldb.nl/other/cursors/).
Other packages distributing this same cursor theme are also don't have a license specified: [AUR](https://aur.archlinux.org/packages/posy-improved-cursors), [opendestktop](https://www.opendesktop.org/p/1795042).

File: maintainers/maintainer-list.nix
Comment (mk3z) [reply maintainers/maintainer-list.nix:1]: What is the recommended way to do this, should I make a new PR?

File: pkgs/data/icons/posy-cursors/default.nix (hunk 5 above)
Comment (nagy) [reply pkgs/data/icons/posy-cursors/default.nix:10]: I mean `hash = "sha256-i0N/QB5uzqHapMCDl6h6PWPJ4GOAyB1ds9qlqmZacLY=";`

File: maintainers/maintainer-list.nix
Comment (nagy) [reply maintainers/maintainer-list.nix:1]: No, you can just remove those parts from that file from this current commit and make a new one here in this PR.

File: pkgs/data/icons/posy-cursors/default.nix (hunk 2 above)
Comment (nagy) [reply pkgs/data/icons/posy-cursors/default.nix:22]: I would say that is a deal breaker then. We don't know if we can redistribute it. Maybe you can ask on the issue tracker of that project for the license?

Comment (pbsds) [reply pkgs/data/icons/posy-cursors/default.nix:25]: It is then by default `unfree`

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -11451,6 +11451,11 @@
     githubId = 5698461;
     name = "Maciej Kazulak";
   };
+  mkez = {
+    github = "mk3z";
+    githubId = 52108954;
+    name = "Matias Zwinger";
+  };
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:11458]: would you be willing to add a second form of contact? Email is common

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -30044,6 +30044,8 @@ with pkgs;
 
   polytopes_db = callPackage ../data/misc/polytopes_db { };
 
+  posy-cursors = callPackage ../data/icons/posy-cursors { };
```
Comment (pbsds) [inline pkgs/top-level/all-packages.nix:30047]: please use pkgs/by-name, for more info please view pkgs/by-name/README.md

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 253172` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>posy-cursors</li>
  </ul>
</details>

**Review by nagy — CHANGES_REQUESTED** [review]
If you merge the commits, such that we have one commit for the maintainer file and one where you init the package, then it's fine by me.

**Comment by pbsds** [comment]
Please rebase rather than merge from master

**Comment by mk3z** [comment]
I'll open a new PR to keep things simple and close this one.

**Comment by nagy** [comment]
> I'll open a new PR to keep things simple and close this one.

You may want to use this opportunity to play with `git rebase -i` https://git-scm.com/docs/git-rebase . That skill may come in handy some time. These are not many commits to tackle that.

**Review by nagy — APPROVED** [review]

**Comment by pbsds** [comment]
Welcome to nixpkgs!
